examples_utils/paperspace_dataset_load_check/dataset_upload_checker.py

```diff
@@ -0,0 +1,108 @@
+# Copyright (c) 2023 Graphcore Ltd. All rights reserved.
+from typing import NamedTuple, Optional, List
+from pathlib import Path
+import os
+import hashlib
+import json
+import logging
+
+
+METADATA_FILENAME = "gradient_dataset_metadata.json"
+
+
+def check_files_match_metadata(dataset_folder: str, compare_hash: bool):
+    dataset_folder = Path(dataset_folder)
+    file_list = sorted(list(f for f in dataset_folder.rglob("*") if f.is_file() and f.name != METADATA_FILENAME))
+    gradient_file_arguments = preprocess_list_of_files(dataset_folder, file_list)
+
+    file_metadata = get_files_metadata(gradient_file_arguments, compare_hash)
+
+    f = open(dataset_folder / METADATA_FILENAME)
+    data = json.load(f)
```

Comment:
Either use a context manager or do
```suggestion
    f = open(dataset_folder / METADATA_FILENAME)
    data = json.loads((dataset_folder / METADATA_FILENAME).read_text())
```